### PR TITLE
Feat: modify base tests

### PR DESCRIPTION
**Pull request by AlexTheWizardL**
### This PR focuses on modifying the base tests with the following objectives:

1. **Resolve Incoherence in Testing Data**: Ensure consistency across the testing datasets, aligning them with the updated logic and expected outcomes.
   
2. **Update `dump()` and `restore()` Interface**: Modify the interface for `dump()` and `restore()` to reflect new requirements or implementation changes, improving clarity and functionality.

3. **Define `BaseTest` Class**: Introduce a `BaseTest` class that will store and manage a common set of data used across different test cases, promoting reusability and reducing redundancy in test setup.

4. **Update Error Handling in Tests**: Revise tests that expect errors to include the error class and error message as arguments. This enhances test readability and ensures precise error handling in expected failure scenarios.

This PR is crucial for enhancing the clarity, maintainability, and accuracy of our Base Tests approach.

@lasuk Please review the changes and provide feedback or approval for merging.

### CODE REVIEW

File: pyledger/ledger_engine.py [hunk 1]
```diff
@@ -200,37 +201,36 @@ def restore_from_zip(self, archive_path: str):
...
                 If `None`, accounts remain unchanged.
             ledger (pd.DataFrame | None): Ledger entries of the restored system.
                 If `None`, ledger remains unchanged.
         """
-        if base_currency is not None:
-            self.base_currency = base_currency
+        if settings is not None:
+            base_currency = settings.get("BASE_CURRENCY", None)
+            if base_currency is not None:
+                self.base_currency = base_currency
```
Comment (lasuk) [inline pyledger/ledger_engine.py:233]: More concise:
```
if settings is not None and "BASE_CURRENCY" in settings:
    self.base_currency = settings["BASE_CURRENCY"]
```

File: tests/base_accounts.py [hunk 2]
```diff
@@ -2,26 +2,27 @@
...
-class BaseTestAccounts(ABC):
+class BaseTestAccounts(BaseTest):
 
     @abstractmethod
     @pytest.fixture
     def ledger(self):
         pass
 
     def test_account_mutators(self, ledger):
+        ledger.restore(vat_codes=self.VAT_CODES.tail(1), accounts=self.ACCOUNTS.tail(1))
```
Comment (lasuk) [inline tests/base_accounts.py:18]: Accessing a specific row of `VAT_CODES` as `tail(1)` makes unnecessary assumptions about the last element of the data frame. A more robust version allows for random order of data frame rows. You can for example accesses an element by its name or simply use all VAT_CODES, the superfluous rows won't hurt.

File: tests/base_test.py [hunk 3]
```diff
@@ -24,9 +26,8 @@
     /Assets, 9993,      EUR,         , Transitory Account EUR
     /Assets, 9994,      USD,         , Transitory Account USD
     /Assets, 9995,      CHF,         , Transitory Account CHF
-    /Assets, 9999,      CHF,   Test  , Test Account with VAT
+    /Assets, 9999,      CHF,         , Test Account with VAT
```
Comment (lasuk) [inline tests/base_test.py:29]: Do we test setting account vat codes somewhere, if the `Test` vat code is left aside here?

File: tests/base_test_dump_restore_clear.py [hunk 4]
```diff
@@ -1,47 +1,44 @@
...
 import pytest
-from abc import ABC, abstractmethod
+from abc import abstractmethod
 from consistent_df import assert_frame_equal
-from .constants import LEDGER_ENTRIES, ACCOUNTS, VAT_CODES
+from .base_test import BaseTest
 
 
-TEST_LEDGER = LEDGER_ENTRIES.head(10)
+LEDGER = BaseTest.LEDGER_ENTRIES.head(7)
```
Comment (lasuk) [inline tests/base_test_dump_restore_clear.py:9]: What is the benefit of limiting the number of entries to 7 (or 10 for that matter)?

**Review by lasuk — COMMENTED** [review]
Thanks for the changes. I left a few comments

File: tests/base_accounts.py (hunk 2 above)
Comment (AlexTheWizardL) [reply tests/base_accounts.py:18]: Agree with you, accessing rows with `.tail()` wasn't a smart move

File: tests/base_test.py (hunk 3 above)
Comment (AlexTheWizardL) [reply tests/base_test.py:29]: Yeah, that was made to the advantage of less interdependent data.
The case you are talking about is covered within `test_account_mutators()`

File: tests/base_test_dump_restore_clear.py (hunk 4 above)
Comment (AlexTheWizardL) [reply tests/base_test_dump_restore_clear.py]: That was made to please the `CashCtrl` limitations but should be dropped, thanks for catching this

**Comment by lasuk** [comment]
Thanks for addressing the review comments! Ready to merge ...

**Pull request merged**